packages/batch-nft-reveal/contracts/NFTCollection.sol

```diff
@@ -0,0 +1,285 @@
+
+    function _formatTokenURI(string memory imageURI)
+        internal
+        pure
+        returns (string memory)
+    {
+        return
+            string(
+                abi.encodePacked(
+                    "data:application/json;base64,",
+                    Base64.encode(
+                        bytes(
+                            abi.encodePacked(
+                                '{"name":"',
+                                "NFT", // You can add whatever name here
+                                '", "description":"Batch-revealed NFT!", "attributes":"", "image":"',
+                                imageURI,
+                                '"}'
+                            )
+                        )
+                    )
+                )
+            );
+    }
+
+    function _generateSVG(uint256 _randomness, bool _metadataCleared)
+        public
+        pure
+        returns (string memory)
+    {
+        string[3] memory parts;
+
+        parts[0] = '<svg xmlns="http://www.w3.org/2000/svg" preserveAspectRatio="xMinYMin meet" viewBox="0 0 350 350"><style>.base { fill: white; font-family: serif; font-size: 8px; }</style><rect width="100%" height="100%" fill="black" /><text x="10" y="20" class="base">';
+
+        if (_metadataCleared) {
+            parts[1] = _toString(_randomness);
+        } else {
+            parts[1] = "No randomness assigned";
+        }
+
+        parts[2] = "</text></svg>";
+
+        return string(abi.encodePacked(parts[0], parts[1], parts[2]));
+    }
+
+    function _svgToImageURI(string memory svg)
+        internal
+        pure
+        returns (string memory)
+    {
+        string memory baseURL = "data:image/svg+xml;base64,";
+        string memory svgBase64Encoded = Base64.encode(bytes(string(abi.encodePacked(svg))));
+        return string(abi.encodePacked(baseURL, svgBase64Encoded));
+    }
+
+    function _canReveal() internal view returns (bool) {
+        uint256 unrevealedCount = totalSupply - revealedCount;
+        if (unrevealedCount == 0) {
+            return false;
+        }
+        bool batchSizeCriteria = false;
+        if (revealBatchSize > 0 && unrevealedCount >= revealBatchSize) {
+            batchSizeCriteria = true;
+        }
+        bool intervalCriteria = false;
+        if (
+            revealInterval > 0 &&
+            block.timestamp - lastRevealed > revealInterval
+        ) {
+            intervalCriteria = true;
+        }
+        return (batchSizeCriteria || intervalCriteria);
+    }
+
+    // VRF
+
+    function revealPendingMetadata() public returns (bytes32 requestId) {
+        if (!_canReveal()) {
+            revert RevealCriteriaNotMet();
+        }
+        if (IERC20(linkToken).balanceOf(address(this)) < linkFee) {
+            revert InsufficientLINK();
+        }
+        requestId = requestRandomness(keyHash, linkFee);
+        pendingReveal = true;
+    }
+
+    function _fullfillRandomnessForMetadata(uint256 randomness) internal {
```

Comment:
A small typo here - fulfill * 
```suggestion
    function _fulfillRandomnessForMetadata(uint256 randomness) internal {
```

This typo was causing me some trouble when I was debugging so if you could fix this that would be nice.